File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.ts
function partitionString(s: string): number {
    let arr = new Array(26).fill(0);
    let count = 0;
    for(let i = 0; i<s.length; i++){
        let n = s.charAt(i).charCodeAt(0) - 97;
        if(arr[n]===0){
            arr[n]++;
        }else{
            count++;
            arr.fill(0);
            i--;
        }
    }
    return count+1;
};

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.cpp
class Solution {
public:
    int partitionString(string s) {
        
        // int arr[26] = {0};
        // int* arr = new int[26];
        vector<int>arr(26,0);
        int count = 0;
        for(int i = 0 ; i<s.size(); i++){
            if(arr[s[i]-97] == 0){
                
                arr[s[i]-97]++;
            }else{
                cout << s[i] << endl;
                fill(arr.begin(),arr.end(),0);
                count++;
                i--;
            }
        }
        return count+1;
    }
};